nav2_bringup/params/nav2_params.yaml

```diff
@@ -452,6 +451,7 @@ docking_server:
       use_external_detection_pose: true
       use_battery_status: false # true
       use_stall_detection: false # true
+      dock_direction: "forward"
```

Comment:
I think we need to make sure to update the configuration guide that this is now a plugin param, not server.

Also, I think the migration guide needs this mentioned as well since its subtle. 

Can we also check for the `dock_backwards` parameter on the server to see if its declared? If so, log a warning at this use is depreciated and use it as part of the plugins